packages/assets-controllers/src/TokensController.ts

```diff
@@ -120,7 +85,6 @@ export interface TokensState extends BaseState {
   allTokens: { [key: string]: { [key: string]: Token[] } };
   allIgnoredTokens: { [key: string]: { [key: string]: string[] } };
   allDetectedTokens: { [key: string]: { [key: string]: Token[] } };
-  suggestedAssets: SuggestedAssetMeta[];
```

Comment:
No longer need for status if we don't want to persist them after restarts.

Reply:
As this was previously persisted, does this warrant a migration to delete this data on upgrade?
Not sure what the best practice is and if we prioritise fewer migrations over dormant data in the user state.

Reply:
How was this decision come to? Not saying I disagree with it. I'm just curious what the thinking was here.

Reply:
The ability to persist pending `wallet_watchAsset` after a restart had no tests to cover that case and it was inadvertently broken in a previous PR.

When we found out about the regression, we reached out, asked if that feature was intentional, accidental or important, so that we could decide between fixing it first. The decision was to not restore that feature and have it behave the same way as other confirmations such as the ones that ask specific permissions (which are also not persisted).

Reply:
@adonesky1 I have discussed this with @danjm and @bschorchit and we didn't thought of any relevant use case. As so we have decided to standardise the pending `wallet_watchAsset` confirmation behaviour (i.e no persisting confirmations after a restart... just like permissions for example. The only exception are transactions). But is there a use case that you can think for persisting pending approvals to add suggested assets?

Reply:
I guess the one reason I can think of to keep suggestedAsset state (though we can probably accomplish this by looking at the approval controller state) is that to avoid duplicate suggestions. This is obviously not about persisting state across reloads. We do not currently prevent duplicate suggestions:
<img width="354" alt="Screenshot 2023-05-16 at 1 09 17 PM" src="https://github.com/MetaMask/core/assets/34557516/a0f8ab41-6741-4e34-a70c-aad2095f0fe5">

But I was thinking maybe we ought to?

Reply:
It'd be easier to rate limit approvals from `wallet_watchAsset` type from `ApprovalController`.